File: 128-LongestConsecutiveSequence/128-LongestConsecutiveSequence.cpp
// Last updated: 2/11/2026, 11:08:32 AM
#include <unordered_set>
using namespace std;

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> st;
        for (int i : nums){
            st.insert(i);
        }


        int res = 0;
        for (int i : st){
            if (st.find(i-1) == st.end()){
                int length = 1;

                while (st.find(i+length) != st.end()){
                    length++;
                }

                if (length > res){
                    res = length;
                }


            }
        }
        return res;
    }
};

File: 128-LongestConsecutiveSequence/128-LongestConsecutiveSequence.py
# Last updated: 2/11/2026, 11:08:34 AM
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        st = set(nums)
        res = 0

        for num in st:
            if (num-1 not in st):
                leng = 1
                while (num + leng in st):
                    leng+=1
                res = max (res, leng)
        return res